From 60f5ac55235d584b1b0c84a9a4205bef874906e0 Mon Sep 17 00:00:00 2001
From: atefganm <atef_anim_1974@yahoo.com>
Date: Fri, 16 Apr 2021 19:50:39 +0200
Subject: [PATCH] fix-fp-version

---
 lib/python/Tools/StbHardware.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/lib/python/Tools/StbHardware.py b/lib/python/Tools/StbHardware.py
index 2c8ff63..d35c82b 100755
--- a/lib/python/Tools/StbHardware.py
+++ b/lib/python/Tools/StbHardware.py
@@ -2,12 +2,16 @@ from os import path
 from fcntl import ioctl
 from struct import pack, unpack
 from time import time, localtime, gmtime
+from enigma import getBoxType
 
 
 def getFPVersion():
 	ret = None
 	try:
-		ret = long(open("/proc/stb/fp/version", "r").read())
+		if getBoxType() in ('dm7080','dm820','dm520','dm525','dm900','dm920'):
+			ret = open("/proc/stb/fp/version", "r").read()
+		else:
+			ret = long(open("/proc/stb/fp/version", "r").read())
 	except IOError:
 		try:
 			fp = open("/dev/dbox/fp0")
-- 
2.25.1

